src/ComponentConcerns/HandlesActions.php

```diff
@@ -20,6 +21,28 @@ public function syncInput($name, $value, $rehash = true)
     {
         $propertyName = $this->beforeFirstDot($name);
 
+        if (version_compare(PHP_VERSION, '7.4', '>=')) {
+            try {
+                $propertyType = (new \ReflectionClass($this))
+                    ->getProperty($propertyName)
+                    ->getType();
+            } catch (\Exception $e) {
+                $propertyType = null;
+            }
+
+            $property = $propertyType ? $propertyType->getName() : null;
```

Comment:
Since PHP 8, `getType()` can either return a `ReflectionNamedType` or `ReflectionUnionType`. The latter containing multiple types. I think it's worth supporting union types, and defaulting the chosen type to the first in the list. Alternatively you could throw an exception if this shouldn't try to guess which type to use?

Reply:
Yeah I agree, union types have no place here. Although the code above needs some tweaking to check if it's union and then throw a useful exception - otherwise it may try to use an array as an object.